### ISSUE TITLE
plot_pd: no geom_line-lines if predictor ordered factor

### ISSUE THREAD

**Issue by stefanherzog** [zmjones/edarf#38]
First of all: **I :heart:  your package!** :-)

When using plot_pd, no lines are drawn if the predictor is an ordered factor. See the interaction plot in this example using the `BreastCancer` data set from `mlbench`: [section 5.2](http://rpubs.com/sherzog/98900).

Or is this a feature and not a bug? That is, is this an assumption in the ggplot2-philosophy that points from ordered factors shouldn't be connected?

This could possibly be addressed using the `group` argument in `aes`? See [here](https://kohske.wordpress.com/2010/12/27/faq-geom_line-doesnt-draw-lines/).

**Comment by zmjones** [comment]
No that is a bug. There is a check somewhere in there for ordering that is apparently broken or somehow the ordered factor is being coerced to a unordered factor. 

My computer has been kaput and being repaired until today and I'm a bit behind on other work, but I will get to this soon. 

Thanks for the report and glad you like the package!

**Comment by stefanherzog** [comment]
Thanks for your thoughts on this!

**Comment by zmjones** [comment]
Totally forgot about this for a while, sorry.

So you are right: by default ggplot2 groups factors (including ordered ones). So even though geom_line is called it just prints the points. The alternative would be to coerce ordered factors to integers. I think then I can probably label the axis using the factor labels. Will work on this when I get a chance. Hopefully the whole thing is a bit more reliable now!

**Comment by stefanherzog** [comment]
Thanks!

**Comment by zmjones** [comment]
So I spent some time with this today and I think it makes sense to just do the easy thing here, which is to cast factors to integers, and draw lines if the factor was ordered. So your example should be fixed now. I don't see an easy way of adding the labels but I may revisit this at a later date when I have some more time. If you wanted to take a crack at it you are more than welcome to. I refactored and simplified things a good bit.
